src/utils/Logger.cpp

```diff
@@ -2,13 +2,19 @@
 
 #include "Logger.h"
 
+#include <memory>
+
 namespace Log {
 
-Logger *global_log;
+std::unique_ptr<Logger> global_log;
 
+// Write to stream
 Logger::Logger(logLevel level, std::ostream *os) :
 	_log_level(level), _msg_log_level(Log::Error),
-	_do_output(true), _filename(""),_log_stream(os),
+	_do_output(true), _filename(""),
+	// std::cout is managed globally,
+	// so do nothing when _log_stream goes out of scope
+	_log_stream(os, [](std::ostream*){/* no-op deleter */}),
```

Comment:
On the one hand nice workaround! On the other, if a stream other than `std::cout` is passed it should be clarified that the logger never takes ownership of the given stream pointer and thus will never delete it. Put this in the constructor and class description.

Reply:
What do you think of another possibility?
We could just get rid of the option of passing a `ostream` and just set `_log_stream` to `std::cout`.
The extra option is nowhere used in our compiled code (maybe in some tools, idk).

```
Logger(logLevel level = Log::Error);
```
and
```
Logger::Logger(logLevel level) :

	_log_stream(&(std::cout), [](std::ostream*){/* no-op deleter */}),

{
```

Reply:
Yes but the option to log to a file is always nice to have and the current implementation is sufficient. I'd just clarify the doc as mentioned above and leave it as it is now.